src/hotspot/share/services/threadService.hpp

```diff
@@ -487,10 +487,11 @@ class JavaThreadInObjectWaitState : public JavaThreadStatusChanger {
   bool _active;
 
  public:
-  JavaThreadInObjectWaitState(JavaThread *java_thread, bool timed) :
+  JavaThreadInObjectWaitState(JavaThread *java_thread, bool timed, bool interruptible) :
     JavaThreadStatusChanger(java_thread,
-                            timed ? JavaThreadStatus::IN_OBJECT_WAIT_TIMED : JavaThreadStatus::IN_OBJECT_WAIT) {
-    if (is_alive()) {
+                            interruptible ? (timed ? JavaThreadStatus::IN_OBJECT_WAIT_TIMED : JavaThreadStatus::IN_OBJECT_WAIT)
+                                          : java_lang_Thread::get_thread_status(java_thread->threadObj())) {
```

Comment:
I think we need some explanatory comments here. It took me a while to understand how interruptible==false caused this to all be a (very inefficient) no-op. Thanks

Reply:
Thanks. I've tried to explain it at line 494. But you are right, I'll add an explaning comment.

Reply:
Fixed now.